src/main/webapp/mentors.js

```diff
@@ -3,7 +3,7 @@
  */
 function getMentors() {
   console.log('entering get mentors function/n');
-  let url = new URL('/mentors', location.protocol + '//' + location.hostname);
+  let url = new URL('/mentors', window.location.protocol + '//' + window.location.hostname);
```

Comment:
It might be good to look into using [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams)